lib/Doctrine/DBAL/Driver/FetchUtils.php

```diff
@@ -57,6 +62,44 @@ public static function fetchAllAssociative(Result $result): array
         return $rows;
     }
 
+    /**
+     * Returns an array containing the values of the first column of the result.
+     *
+     * @return array<mixed,mixed>
+     *
+     * @throws Exception
+     */
+    public static function fetchAllAssociativeIndexed(Result $result): array
```

Comment:
I'd rather have these methods implemented in a separate class under the `ForwardCompatibility` namespace. This way, it will be clear that it won't need to be merged up. Also, it reduces the risk of accidentally introducing an API that isn't supported in `3.x`.

Reply:
Added methods are used exactly in the same way that previously existing ones : fetchOne(), fetchAllNumeric(), fetchAllAssociative() ... etc. Wouldn't it be weird to split them in 2 separate classes ?

Reply:
The point is that drivers must implement only a subset of the methods while the ones you're adding are implemented at the wrapper level. You're adding wrapper-specific methods to the driver level.